From 7292ca078fd3f5c6dc8256a7a2ecc9f3d38935ef Mon Sep 17 00:00:00 2001
From: Marc Kleine-Budde <mkl@pengutronix.de>
Date: Fri, 6 Nov 2009 09:26:56 +0100
Subject: [PATCH 5/5] [mkfs_vfat] silence errors on BE platforms

ilence this warning, which crashes when CONFIG_WERROR is active:

cc1: warnings being treated as errors
util-linux/mkfs_vfat.c: In function 'mkfs_vfat_main':
util-linux/mkfs_vfat.c:468: error: large integer implicitly truncated to unsigned type

Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>
---
 util-linux/mkfs_vfat.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/util-linux/mkfs_vfat.c b/util-linux/mkfs_vfat.c
index bdd4dd8..ade2282 100644
--- a/util-linux/mkfs_vfat.c
+++ b/util-linux/mkfs_vfat.c
@@ -174,11 +174,11 @@ void BUG_unsupported_field_size(void);
 #define STORE_LE(field, value) \
 do { \
 	if (sizeof(field) == 4) \
-		field = cpu_to_le32(value); \
+		field = (typeof(field))cpu_to_le32(value);	\
 	else if (sizeof(field) == 2) \
-		field = cpu_to_le16(value); \
+		field = (typeof(field))cpu_to_le16(value); \
 	else if (sizeof(field) == 1) \
-		field = (value); \
+		field = (typeof(field))(value); \
 	else \
 		BUG_unsupported_field_size(); \
 } while (0)
-- 
1.6.5.2

